File: typescript/src/types/keys.ts
export interface APIKey {
  uid: string;
  name: string;
  prefix: string;
  status: "active" | "suspended" | "revoked";
  rpm_limit?: number;
  created_at: string;
  last_used_at?: string | null;
  /** Only present on creation */
  key?: string;
}

export interface APIKeyList {
  items: APIKey[];
  total: number;
}

export interface APIKeyCreateParams {
  name: string;
  rpm_limit?: number;
}

export interface APIKeyUpdateParams {
  name?: string;
  rpm_limit?: number;
}

File: python/src/azex/types/keys.py
"""Types for the API keys resource."""

from __future__ import annotations

from typing import List, Literal, Optional

from pydantic import BaseModel


class APIKey(BaseModel):
    uid: str
    name: str
    prefix: str
    status: Literal["active", "suspended", "revoked"]
    rpm_limit: Optional[int] = None
    created_at: str
    last_used_at: Optional[str] = None
    # Only present on creation
    key: Optional[str] = None


class APIKeyList(BaseModel):
    items: List[APIKey]
    total: int
